File: app/src/main/java/eu/sesma/dagger/platform/ActivityModule.java
package eu.sesma.dagger.platform;

import eu.sesma.dagger.di.PerActivity;
import eu.sesma.dagger.ui.ActCollaborator;

import dagger.Module;
import dagger.Provides;

@Module
public class ActivityModule {

    @Provides
    @PerActivity
    ActCollaborator provideActCollaborator() {
        return new ActCollaborator();
    }
}

File: app/src/main/java/eu/sesma/dagger/platform/ActivityModule.kt
package eu.sesma.dagger.platform

import dagger.Module
import dagger.Provides
import eu.sesma.dagger.di.ActivityScoped
import eu.sesma.dagger.ui.ActCollaborator
import eu.sesma.dagger.ui.BaseActivity

@Module
class ActivityModule(private val activity: BaseActivity) {

    @Provides
    @ActivityScoped
    fun provideActivity(): BaseActivity {
        return this.activity
    }

    @Provides
    @ActivityScoped
    fun provideActCollaborator(): ActCollaborator {
        return ActCollaborator()
    }
}
